### PR TITLE
ARROW-72 Run performance tests on prototype and evaluate our approach

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: bindings/python/benchmark.py [hunk 1]
```diff
@@ -143,6 +152,17 @@ def to_arrow(use_large):
     find_arrow_all(c, {}, schema=schema)
 
 
+@bench("arrow-roundtrip")
+def roundtrip_arrow(use_large):
```
Comment (ShaneHarvey) [inline bindings/python/benchmark.py:156]: "roundtrip" to me implies read + write while this benchmark only cares about the write. Let's rename to "conventional":
```
@bench("insert-arrow")
def insert_arrow(use_large):
...
@bench("insert-arrow-conventional")
def insert_arrow_conventional(use_large):
```

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/benchmark.py [hunk 2]
```diff
@@ -143,6 +152,17 @@ def to_arrow(use_large):
     find_arrow_all(c, {}, schema=schema)
 
 
+@bench("insert_arrow")
+def insert_arrow(use_large):
+    write(db[collection_names[use_large]], arrow_tables[use_large])
+
+
+@bench("insert_naive")
+def insert_naive(use_large):
```
Comment (ShaneHarvey) [inline bindings/python/benchmark.py:161]: naive->conventional (the term the rest of the file uses).

File: bindings/python/pymongoarrow/api.py [hunk 3]
```diff
@@ -279,23 +279,24 @@ def write(collection, tabular):
...
             enc_tab = RawBSONDocument(
                 encode(next(tabular_gen), codec_options=collection.codec_options)
             )
             cur_batch.append(enc_tab)
-            cur_size += len(enc_tab)
+            cur_size += len(enc_tab.raw)
             i += 1
         try:
-            collection.insert_many(cur_batch)
+            collection.insert_many(cur_batch, bypass_document_validation=True)
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/api.py:299]: Why add bypass_document_validation=True?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/pymongoarrow/api.py (hunk 3 above)
Comment (juliusgeo) [reply bindings/python/pymongoarrow/api.py]: That was in error. Removed.

**Review by ShaneHarvey — APPROVED** [review]
LGTM!

**Pull request merged**